crates/bevy_animation/src/lib.rs

```diff
@@ -691,217 +740,308 @@ where
         + tangent_in_end * step_duration * (lerp.powi(3) - lerp.powi(2))
 }
 
-#[allow(clippy::too_many_arguments)]
-fn apply_animation(
-    weight: f32,
+fn update_transitions(player: &mut AnimationPlayer, time: &Time) {
+    player.transitions.retain_mut(|animation| {
+        animation.current_weight -= animation.weight_decline_per_sec * time.delta_seconds();
+        animation.current_weight > 0.0
+    });
+}
+
+/// Adds animation support to an app
+#[derive(Default)]
+pub struct AnimationPlugin;
+
+impl Plugin for AnimationPlugin {
+    fn build(&self, app: &mut App) {
+        app.init_asset::<AnimationClip>()
+            .register_asset_reflect::<AnimationClip>()
+            .register_type::<AnimationPlayer>()
+            .add_systems(Update, update_animation_targets)
+            .add_systems(
+                PostUpdate,
+                (animation_player, animate_targets)
+                    .chain()
+                    .before(TransformSystem::TransformPropagate),
+            );
+    }
+}
+
+impl AnimationTarget {
+    // For private use in `update_animation_targets` only.
+    fn new() -> AnimationTarget {
+        AnimationTarget {
```

Comment:
Same here. I agree it's a matter of choice, but since this is a built-in crate, it's probably better to keep things consistent with the pre-existing convention.

```suggestion
        Self {
```